tests/test_web_protocol.py

```diff
@@ -338,6 +341,11 @@ def close():
         "Error handling request", exc_info=mock.ANY)
 
 
+@pytest.mark.xfail(
```

Comment:
Please replace `xfail` with `skipif`, apply for Python 3.5 only.
I'd like to make the test passing on Python 3.6+ at least

Reply:
In my previous commit it was ` and IS_PY35` but then I noticed that this is unstable and happens under other versions as well.
I think it must be xfail because of this. And I need to fix the reason instead.

Reply:
The reason is OpenSSL version bundled with Python. Have no idea how to "fix" it.

Reply:
This test doesn't seem related to TLS in any way. You're probably confusing it with the other one which actually was failing because of OpenSSL and I've patched that up.